hydropandas/extensions/plots.py

```diff
@@ -691,13 +714,37 @@ def series_per_group(self, plot_column, by=None, savefig=True, outputdir="."):
             ax.set_ylabel(row["unit"])
             ax.grid(True)
             f.tight_layout()
+
             if savefig:
-                if isinstance(row.obs, GroundwaterObs):
-                    name = name.split("-")[0]
+                if isinstance(by, list):
+                    by_name = "-".join(by)
+                else:
+                    by_name = by
+                    groupname = "-".join(groupname)
+                if naming_method is None:
+                    filename = f"series_by_{by_name}_group_{groupname}.png"
+                elif naming_method == "infer_name_monitoring_well":
+                    if "-" in name:
+                        filename = f"{name.split('-')[0]}.png"
+                    elif "_" in name:
+                        filename = f"{name.split('_')[0]}.png"
+                    else:
+                        filename = f"{name}.png"
+                elif isinstance(naming_method, str):
+                    filename = (
+                        f"series_by_{by_name}_group_"
+                        f"{'-'.join(group[naming_method].unique().tolist())}.png"
+                    )
+
                 f.savefig(
-                    os.path.join(outputdir, f"{name}.png"), bbox_inches="tight", dpi=150
+                    os.path.join(outputdir, filename), bbox_inches="tight", dpi=150
```

Comment:
Is `bbox_inches="tight"` required, because `f.tight_layout()` is applied as well (line 716)

Reply:
I'm not sure, i usually use both...

Reply:
The tight_layout is for the figure layout and bbox_inches recalculates the figure extent before saving. In an ideal world those would be identical, but I'm not sure they are in practice. And it doesnt hurt to leave them both in.

And for what its worth, I see matplotlib actually mildly discourages tight_layout and suggests using constrained layout these days:
![image](https://github.com/ArtesiaWater/hydropandas/assets/6552225/ccba7374-3d79-4672-a12c-e1afd6fcc610)